internal/sanitize.go

```diff
@@ -40,6 +42,32 @@ func Sanitize(s string) string {
 	return s
 }
 
+var (
+	smu       sync.Mutex
+	sanitized = make(map[string]string)
+)
+
+// SanitizeNoClash returns an error if in's sanitization clashes
```

Comment:
We should modify the sanitized string rather than returning an error. At least, we should try to do a few iterations and trying to find a properly sanitized combination.

I'd not spend much time on this PR right now, the sanitization strategy should be implemented similarly at all other languages for consistency and requires a spec.